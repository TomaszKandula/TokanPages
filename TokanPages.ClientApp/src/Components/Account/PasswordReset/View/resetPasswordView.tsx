import * as React from "react";
import { ViewProperties } from "../../../../Shared/Abstractions";
import { ReactChangeEvent, ReactKeyboardEvent } from "../../../../Shared/types";
import { Icon, ProgressBar, Skeleton, TextField } from "../../../../Shared/Components";
import { PasswordResetProps } from "../passwordReset";

interface Properties extends ViewProperties, PasswordResetProps {
    progress: boolean;
    caption: string;
    button: string;
    email: string;
    keyHandler: (event: ReactKeyboardEvent) => void;
    formHandler: (event: ReactChangeEvent) => void;
    buttonHandler: () => void;
    labelEmail: string;
}

const ActiveButton = (props: Properties): React.ReactElement => {
    return (
        <button
            type="submit"
            onClick={props.buttonHandler}
            className="bulma-button bulma-is-link bulma-is-light bulma-is-fullwidth"
            disabled={props.progress || props.email.length === 0}
        >
            {!props.progress ? props.button : <ProgressBar size={20} />}
        </button>
    );
};

export const PasswordResetView = (props: Properties): React.ReactElement => {
    return (
        <section className={props.background}>
            <div className="bulma-container bulma-is-max-tablet">
                <div className={!props.className ? "py-6" : props.className}>
                    <div className="bulma-card">
                        <div className="bulma-card-content">
                            <Skeleton isLoading={props.isLoading}>
                                <div className="is-flex is-flex-direction-column is-align-items-center">
                                    <Icon name="AccountCircle" size={3} className="has-text-link" />
                                    <p className="is-size-3 has-text-grey">{props.caption}</p>
                                </div>
                                <div className="my-5">
                                    <TextField
                                        required
                                        uuid="email"
                                        autoComplete="email"
                                        onKeyUp={props.keyHandler}
                                        onChange={props.formHandler}
                                        value={props.email}
                                        placeholder={props.labelEmail}
                                    />
                                </div>
                                <div className="my-5">
                                    <ActiveButton {...props} />
                                </div>
                            </Skeleton>
                        </div>
                    </div>
                </div>
            </div>
        </section>
    );
};
