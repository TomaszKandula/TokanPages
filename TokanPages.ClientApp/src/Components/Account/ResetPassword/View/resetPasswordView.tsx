import * as React from "react";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import { Card, CardContent, CircularProgress } from "@material-ui/core";
import { AccountCircle } from "@material-ui/icons";
import Skeleton from "@material-ui/lab/Skeleton";
import { ViewProperties } from "../../../../Shared/Abstractions";
import { ReactChangeEvent, ReactKeyboardEvent } from "../../../../Shared/types";
import { ResetPasswordProps } from "../resetPassword";
import { ResetPasswordStyle } from "./resetPasswordStyle";

interface Properties extends ViewProperties, ResetPasswordProps {
    progress: boolean;
    caption: string;
    button: string;
    email: string;
    keyHandler: (event: ReactKeyboardEvent) => void;
    formHandler: (event: ReactChangeEvent) => void;
    buttonHandler: () => void;
    labelEmail: string;
}

const ActiveButton = (props: Properties): React.ReactElement => {
    const classes = ResetPasswordStyle();
    return (
        <Button
            fullWidth
            type="submit"
            variant="contained"
            onClick={props.buttonHandler}
            className={classes.button}
            disabled={props.progress}
        >
            {!props.progress ? props.button : <CircularProgress size={20} />}
        </Button>
    );
};

export const ResetPasswordView = (props: Properties): React.ReactElement => {
    const classes = ResetPasswordStyle();
    return (
        <section className={classes.section} style={props.background}>
            <Container className={classes.container}>
                <Box pt={props.pt ?? 18} pb={props.pb ?? 10}>
                    <Card elevation={0} className={classes.card}>
                        <CardContent className={classes.card_content}>
                            <Box mb={3} textAlign="center">
                                <AccountCircle className={classes.account} />
                                <Typography className={classes.caption}>
                                    {props.isLoading ? <Skeleton variant="text" /> : props.caption}
                                </Typography>
                            </Box>
                            <Box>
                                <Grid container spacing={2}>
                                    <Grid item xs={12}>
                                        {props.isLoading ? (
                                            <Skeleton variant="rect" width="100%" height="45px" />
                                        ) : (
                                            <TextField
                                                required
                                                fullWidth
                                                id="email"
                                                name="email"
                                                variant="outlined"
                                                autoComplete="email"
                                                onKeyUp={props.keyHandler}
                                                onChange={props.formHandler}
                                                value={props.email}
                                                label={props.labelEmail}
                                            />
                                        )}
                                    </Grid>
                                </Grid>
                                <Box my={2}>
                                    {props.isLoading ? (
                                        <Skeleton variant="rect" width="100%" height="40px" />
                                    ) : (
                                        <ActiveButton {...props} />
                                    )}
                                </Box>
                            </Box>
                        </CardContent>
                    </Card>
                </Box>
            </Container>
        </section>
    );
};
