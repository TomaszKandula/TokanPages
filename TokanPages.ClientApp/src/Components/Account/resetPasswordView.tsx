import * as React from "react";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import { Card, CardContent } from "@material-ui/core";
import { AccountCircle } from "@material-ui/icons";
import Skeleton from "@material-ui/lab/Skeleton";
import { IGetResetPasswordContent } from "../../Redux/States/Content/getResetPasswordContentState";
import resetPasswordStyle from "./Styles/resetPasswordStyle";

export default function ResetPasswordView(props: IGetResetPasswordContent) 
{
    const classes = resetPasswordStyle();
    return (
        <section>
            <Container maxWidth="sm">
                <Box pt={18} pb={10}>             
                    <Card elevation={4}>
                        <CardContent className={classes.card}>
                            <Box mb={3} textAlign="center">
                                <AccountCircle color="primary" style={{ fontSize: 72 }} />
                                <Typography variant="h5" component="h2" color="textSecondary">
                                    {props.isLoading ? <Skeleton variant="text" /> : props.content?.caption}
                                </Typography>
                            </Box>
                            <Box>
                                <Grid container spacing={2}>
                                    <Grid item xs={12}>
                                        <TextField 
                                            variant="outlined" required fullWidth 
                                            name="email" id="email" label="Email address" autoComplete="email" 
                                        />
                                    </Grid>
                                </Grid>
                                <Box my={2}>
                                    <Button fullWidth variant="contained" color="primary">
                                        {props.isLoading ? <Skeleton variant="text" /> : props.content?.button}
                                    </Button>
                                </Box>
                            </Box>
                        </CardContent>   
                    </Card>                    
                </Box>
            </Container>
        </section>
    );
}
