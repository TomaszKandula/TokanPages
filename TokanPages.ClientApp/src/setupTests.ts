// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
