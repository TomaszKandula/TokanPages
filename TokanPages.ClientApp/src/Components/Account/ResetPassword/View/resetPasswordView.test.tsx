import "../../../../setupTests";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "enzyme";
import { ResetPasswordView } from "./resetPasswordView";

describe("test account group component: resetPasswordView", () => {
    it("should render correctly '<ResetPasswordView />' when content is loaded.", () => {
        const html = render(
            <BrowserRouter>
                <ResetPasswordView
                    isLoading={false}
                    progress={false}
                    caption="Reset Password"
                    button="Reset"
                    email="tokan@dfds.com"
                    keyHandler={jest.fn()}
                    formHandler={jest.fn()}
                    buttonHandler={jest.fn()}
                    labelEmail="Email address"
                />
            </BrowserRouter>
        );

        expect(html).toMatchSnapshot();
    });
});
