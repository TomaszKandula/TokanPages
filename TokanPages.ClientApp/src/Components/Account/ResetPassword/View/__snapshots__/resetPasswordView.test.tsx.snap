// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test account group component: resetPasswordView. Renders correctly '<ResetPasswordView />' when content is loaded. 1`] = `
<section
  className="makeStyles-section-1"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="sm"
  >
    <Styled(MuiBox)
      pb={10}
      pt={18}
    >
      <WithStyles(ForwardRef(Card))
        className="makeStyles-card-4"
        elevation={0}
      >
        <WithStyles(ForwardRef(CardContent))
          className="makeStyles-card_content-5"
        >
          <Styled(MuiBox)
            mb={3}
            textAlign="center"
          >
            <Memo(ForwardRef(AccountCircleIcon))
              className="makeStyles-account-2"
            />
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-caption-3"
            >
              Password Reset
            </WithStyles(ForwardRef(Typography))>
          </Styled(MuiBox)>
          <Styled(MuiBox)>
            <WithStyles(ForwardRef(Grid))
              container={true}
              spacing={2}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={12}
              >
                <WithStyles(ForwardRef(TextField))
                  autoComplete="email"
                  fullWidth={true}
                  id="email"
                  label="Email address"
                  name="email"
                  onChange={[MockFunction]}
                  required={true}
                  value="tokan@dfds.com"
                  variant="outlined"
                />
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
            <Styled(MuiBox)
              my={2}
            >
              <ActiveButton />
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(Card))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Container))>
</section>
`;
